src/server/gc_worker/gc_worker.rs

```diff
@@ -1483,6 +1519,118 @@ pub mod test_gc_worker {
             Ok(self.0.into())
         }
     }
+
+    #[derive(Clone)]
+    pub struct MultiRocksEngine {
+        // Factory is not a normal way to fetch tablet and is just used in test to ease the test.
+        // Note: at most one tablet is allowed to exist for each region in the cache of factory
+        pub factory: Arc<TestTabletFactoryV2>,
+        pub region_info: HashMap<u64, Region>,
+    }
+
+    impl Engine for MultiRocksEngine {
+        type Snap = RegionSnapshot<RocksSnapshot>;
+        type Local = RocksEngine;
+
+        fn multi_rocks(&self) -> bool {
+            true
+        }
+
+        fn kv_engine(&self) -> Self::Local {
+            panic!()
+        }
+
+        fn modify_on_kv_engine(
+            &self,
+            mut region_modifies: HashMap<u64, Vec<Modify>>,
+        ) -> kv::Result<()> {
+            for modifies in region_modifies.values_mut() {
```

Comment:
Why loop twice?

Reply:
One for region id -> Vec<Modify>, and the other is for Modify in vector.

Reply:
Why not
```
for (region_id, modifies) in region_modifies {
    for modify in &mut modifies {
        xxx
    }
    write_modifies(factory.get_tablet(region_id), modifies);
}
```